### PR TITLE
CORE-17007 : Changes for interop p2p params

### PR DESCRIPTION
**Pull request by SaurabhR3**
1. Added logic to fetch p2p params for interop
2. Reverted group policy interface
3. Added/Fixed Test

### CODE REVIEW

File: components/membership/group-policy-impl/src/main/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImpl.kt [hunk 1]
```diff
@@ -103,7 +104,20 @@ class GroupPolicyProviderImpl @Activate constructor(
         listeners.put(name, listener)?.stop()
     }
 
-    override fun getP2PParameters(holdingIdentity: HoldingIdentity) = getGroupPolicy(holdingIdentity)?.p2pParameters
+    override fun getP2PParameters(holdingIdentity: HoldingIdentity) : GroupPolicy.P2PParameters? {
+        val mgmGroupPolicy = getGroupPolicy(holdingIdentity)
+        return if(mgmGroupPolicy == null ) {
+            val groupPolicyJson = interopGroupPolicyReadService.getGroupPolicy(holdingIdentity.groupId)
+            if(groupPolicyJson != null) {
```
Comment (ac101m) [inline components/membership/group-policy-impl/src/main/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImpl.kt]: Code style, space following if

**Review by ac101m — CHANGES_REQUESTED** [review]
Couple trivial things, otherwise looks good to me at a glance. Probably good to get someone with more familiarity to review this.

**Review by ac101m — APPROVED** [review]
LGTM

**Review by wzur-r3 — COMMENTED** [review]
LGTM for files owned by BLT (Gradle configurations)

**Review by wzur-r3 — APPROVED** [review]

File: components/link-manager/build.gradle [hunk 2]
```diff
@@ -45,11 +45,15 @@ dependencies {
     implementation project(":components:membership:membership-persistence-client")
     implementation project(":libs:metrics")
     implementation project(":libs:tracing")
+//    implementation project(":components:interop:interop-group-policy-read-service")
```
Comment (jmacmahonr3) [inline components/link-manager/build.gradle]: Delete commented lines.

File: components/link-manager/build.gradle [hunk 3]
```diff
@@ -45,11 +45,15 @@ dependencies {
...
+//    implementation project(":components:interop:interop-group-policy-read-service-impl")
 
     runtimeOnly "org.apache.felix:org.apache.felix.scr:$felixScrVersion"
     runtimeOnly "org.osgi:org.osgi.service.component:$osgiServiceComponentVersion"
     runtimeOnly "org.osgi:org.osgi.util.function:$osgiUtilFunctionVersion"
     runtimeOnly "org.osgi:org.osgi.util.promise:$osgiUtilPromiseVersion"
+//    runtimeOnly project(":components:interop:interop-group-policy-read-service")
```
Comment (jmacmahonr3) [inline components/link-manager/build.gradle]: Delete

File: components/membership/group-policy-impl/src/test/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImplTest.kt [hunk 4]
```diff
@@ -1031,14 +1053,12 @@ class GroupPolicyProviderImplTest {
...
-    fun `Interop groupPolicy is returned when CPI metadata is null `() {
-        startComponentAndDependencies()
-        postConfigChangedEvent()
-        assertExpectedGroupPolicy(
-            groupPolicyProvider.getGroupPolicy(holdingIdentity6),
-            groupId3,
-            regProtocol3
-        )
-    }
+//    @Test
```
Comment (jmacmahonr3) [inline components/membership/group-policy-impl/src/test/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImplTest.kt]: Is this test failing ? 
Delete or rewrite

**Review by jmacmahonr3 — CHANGES_REQUESTED** [review]
Minor Comments but once resolved looks good

File: components/membership/group-policy-impl/src/main/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImpl.kt (hunk 1 above)
Comment (SaurabhR3) [reply components/membership/group-policy-impl/src/main/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImpl.kt]: Fixed in 2e92d71

File: components/link-manager/build.gradle (hunk 2 above)
Comment (SaurabhR3) [reply components/link-manager/build.gradle]: Fixed in 4c49eef

File: components/link-manager/build.gradle (hunk 3 above)
Comment (SaurabhR3) [reply components/link-manager/build.gradle]: Fixed in 4c49eef

File: components/membership/group-policy-impl/src/test/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImplTest.kt (hunk 4 above)
Comment (SaurabhR3) [reply components/membership/group-policy-impl/src/test/kotlin/net/corda/membership/impl/grouppolicy/GroupPolicyProviderImplTest.kt]: No, it was commented to check something
Fixed in 4c49eef

**Review by jmacmahonr3 — APPROVED** [review]
LGTM

**Pull request merged**